src/api/controllers/db-controller.ts

```diff
@@ -468,34 +478,58 @@ export interface GetTxWithEventsArgs extends GetTxArgs {
   eventOffset: number;
 }
 
+function isDroppedTx(tx: DbMempoolTx): boolean {
+  switch (tx.status) {
+    case DbTxStatus.DroppedReplaceByFee:
+    case DbTxStatus.DroppedReplaceAcrossFork:
+    case DbTxStatus.DroppedTooExpensive:
+    case DbTxStatus.DroppedStaleGarbageCollect:
+      return true;
+    default:
+      return false;
+  }
+}
+
 export async function getTxFromDataStore(
   db: DataStore,
   args: GetTxArgs | GetTxWithEventsArgs
 ): Promise<FoundOrNot<Transaction>> {
   let dbTx: DbTx | DbMempoolTx;
   let dbTxEvents: DbEvent[] = [];
-  // First check mempool
-  const mempoolTxQuery = await db.getMempoolTx(args.txId);
-  if (mempoolTxQuery.found) {
+
+  const txQuery = await db.getTx(args.txId);
+  const mempoolTxQuery = await db.getMempoolTx({ txId: args.txId, includePruned: true });
+
+  // If mined.canonical = false && mempool.dropped = true then use mempool
+  if (
+    txQuery.found &&
+    mempoolTxQuery.found &&
+    !txQuery.result.canonical &&
+    isDroppedTx(mempoolTxQuery.result)
+  ) {
     dbTx = mempoolTxQuery.result;
-  } else {
-    const txQuery = await db.getTx(args.txId);
-    if (!txQuery.found) {
-      return { found: false };
-    }
+  } else if (txQuery.found) {
```

Comment:
I think you want something like:

```
if ( txQuery.found && txQuery.result.canonical ) {
      dbTx = txQuery.result;
} else {
      dbTx = mempoolTxQuery.result;
}
```

Where you use the "mempool" tx if the tx hasn't been mined in a canonical block. However, this brings up a question-- what happens when a tx is remined in a fork? What would `getTx` return in that case?

Reply:
I refactored and simplified the conditions here. The only bit of weirdness is when a tx only exists as non-canonical, and missing from the mempool table completely. I'm not sure if a mempool event is guaranteed for every tx, for example when syncing from scratch. In this case, it just returns the mined tx but with the `canonical: false`. The explorer should already have code to handle this.

 I added a log warning when this happens, although it seems like this may be a rare but expected condition?

Reply:
> I'm not sure if a mempool event is guaranteed for every tx, for example when syncing from scratch. In this case, it just returns the mined tx but with the canonical: false flag. 

Yep, that's exactly right.

> I added a log warning when this happens, although, it seems like this may be an expected (albeit rare) condition?

Yep

Reply:
Okay 👍, in that case, do you think this PR is ready to merge?

Reply:
Yep, LGTM